impeller/aiks/canvas.h

```diff
@@ -160,9 +160,9 @@ class Canvas {
       const Size& corner_radii,
       Entity::ClipOperation clip_op = Entity::ClipOperation::kIntersect);
 
-  void DrawTextFrame(const std::shared_ptr<TextFrame>& text_frame,
-                     Point position,
-                     const Paint& paint);
+  virtual void DrawTextFrame(const std::shared_ptr<TextFrame>& text_frame,
+                             Point position,
+                             const Paint& paint);
```

Comment:
I think the shared data and logic between the old and new canvas is going to get confusing real fast as we continue ramping up the new dispatcher.

Why not just make the new "Canvas" not share an interface with the original Canvas at all? That way, we can start diverging immediately where it makes sense, and we can incorporate @flar's changes in the new Canvas without having to mess with the original interface unnecessarily.

Reply:
This was your suggestion to make the migration more gradual without fully rewriting the renderer :)

Reply:
My suggestion was more to reuse Entities render methods, not the Aiks or Dispatcher logic. I think not extending Aiks would be a minor change here, since it doesn't look like you're actually using much Aiks stuff (unless I'm missing something).

Reply:
Oh I was never going to rewrite entities, that s waaay too much work for one patch. I could do a one time copy/paste of the dispatcher and canvas logic but its quite non-trivial and I wouldn't want to maintain that for too long. If you are able to help me get this work finished quickly I think that would be OK, but I wouldn't want to maintain two dispatchers for very long.

Reply:
Ah, I totally misread this patch -- I just noticed that you're overriding `ExperimentalCanvas::AddEntityToCurrentPass` to call the render method, and it actually _is_ using most of the original Canvas logic. Disregard. 😅

Reply:
`current_pass_` and `base_pass_` aren't really needed for the experimental canvas. It would be nice if we renamed "Canvas" to "CanvasBase" and then Canvas and ExperimentalCanvas would subclass from that with their differences and associated data hidden in their own sub-class...?